src/NUnitFramework/tests/Internal/Filters/FullNameFilterTests.cs

```diff
@@ -61,13 +62,51 @@ public void PassTest()
             Assert.False(_filter.Pass(_anotherFixture));
         }
 
+        [Test]
         public void ExplicitMatchTest()
         {
             Assert.That(_filter.IsExplicitMatch(_topLevelSuite));
             Assert.That(_filter.IsExplicitMatch(_dummyFixture));
-            Assert.False(_filter.IsExplicitMatch(_dummyFixture.Tests[0]));
+            if (((FullNameFilter)_filter).IsRegex)
+                Assert.That(_filter.IsExplicitMatch(_dummyFixture.Tests[0]));
```

Comment:
I had to change this to work, but even after reverting my changes, I don't see how this test ever worked with the regex test case. Any thoughts?